src/main/java/it/eng/idsa/businesslogic/service/impl/CommunicationServiceImpl.java

```diff
@@ -71,20 +71,20 @@ public String sendData(String endpoint, String data) {
 	}
 
 	@Override
-	public String sendDataAsJson(String endpoint, String data) {
+	public String sendDataAsJson(String endpoint, String data, String contentType) {
 		RestTemplate restTemplate = new RestTemplate();
 
 		HttpHeaders headers = new HttpHeaders();
-		headers.set("Content-Type", "application/ld+json;charset=UTF-8");
+		headers.set("Content-Type", contentType);
 
 		HttpEntity<String> entity = new HttpEntity<>(data, headers);
 		
 		ResponseEntity<String> result;
 		try {
 			result = restTemplate.exchange(endpoint, HttpMethod.POST, entity, String.class);
-		} catch (RestClientException e) {
-			logger.error("Error while making a request", e);
-			return null;
+		} catch (Exception e) {
```

Comment:
Why just logging exception here and rethrow it after?
Why not let exception pass to upper layer and handle it then?

Reply:
I now that I changed this but seems that I have somehow lost the changes. The exceptions swap remained by mistake during some changes of the junit tests. I will change this